pkg/lang/javascript/plugin_pubsub.go

```diff
@@ -360,7 +375,14 @@ func (p *Pubsub) generateEmitterDefinitions() (err error) {
 }
 
 func findTopics(f *core.SourceFile, spec VarSpec, query string, methodName string) (topics []string) {
-	varName := findVarName(f, spec)
+	relPath, _ := filepath.Rel(filepath.Dir(f.Path()), spec.DefinedIn)
+	newSpec := VarSpec{
+		DefinedIn:    relPath,
+		InternalName: spec.InternalName,
+		VarName:      spec.VarName,
+	}
+
+	varName := findVarName(f, newSpec)
```

Comment:
This seems like a shortcoming of `findVarName` that should be fixed inside, not outside.

Reply:
will move this logic in there and panic on the relPAth error. Otherwise we need a major refactor of finding every method that uses findVarName

Reply:
Realistically, relPath only errors when one of the paths is absolute, which we should never have. The `panic` is like an assert that makes sure that's true.

Reply:
It's been a while, so I forget exactly what all the inputs to `findVarMean` — but doesn't moving `relPath, err := filepath.Rel(filepath.Dir(f.Path()), spec.DefinedIn)` into it make it more language-specific? There are languages where relative imports aren't a thing.

Reply:
It is indeed language specific, but the var finder is in the javascript package so I'd expect it to be.

Reply:
🤦‍♂️ So it is. For some reason I thought it was in a language-agnostic package (and I didn't even think to double check that assumption in the path). Sorry — carry on! :)

Reply:
yeah i just realized, wouldnt this cause issues for golang?

Reply:
oh wait this is js specific